src/Model/Action.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\core\DIContainerTrait;
+use atk4\core\Exception;
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+use atk4\data\Model;
+
+/**
+ * Implements generic user action. Assigned to a model it can be invoked by a user. Model\Action class contains a
+ * meta information about the action (arguments, permissions, scope, etc) that will help UI/API or add-ons to display
+ * action trigger (button) correctly in an automated way.
+ *
+ * Action must NOT rely on any specific UI implementation.
+ *
+ * @property Model $owner
+ */
+class Action
```

Comment:
Does actions makes sense for backend only? (or are they user UI/GUI helpers)

What is your reasoning to move it under Model?

Basically with this approach we will move everything under Model :)

Reply:
Action is backend and UI helpers.

Namespaces will be organized as per hierarchy:
folder `Model\Scope`
class `Model\Action`
class `Model\Field`
class `Model\Field\Boolean`
class `Model\Field\Callback`, etc

The persistence part will be under `Persistence` 
`Persistence\SQL`
`Persistence\SQL\Join`
`Persistence\SQL\Field\Expression`
`Persistence\SQL\Reference\HasOne`
`Persistence\Array_`
`Persistence\Array_\Join`
`Persistence\Static_`, etc

The rest under `Util`/`Utils`

So design is: class + subfolder (with same name) for specific/extending classes
This is my intended universal NS design over all repos (as already done in atk4\dsql)

Reply:
And Field_SQL?

Reply:
If necessary it will be `Persistence\SQL\Field` (semantic for Field specific to SQL Persistence) but it is marked deprecated so I have bothered not moving it.

Reply:
This seems good. Can you finish the renaming of DSQL to strict PascalCase, ie. Mysql etc.?

Reply:
https://www.php-fig.org/psr/psr-7/ check for example this PSR from PHP and notice, they do not use `HTTP` or `URI`, but instead `Http` and `Uri`.

Reply:
How about MySql, Postgre, etc (to be in the middle, but I doubt you will agree 😄 )
With your logic we shall then have now HasOne but Hasone in  atk4\data?

Reply:
No! Mysql. Would you write my_sql? No magic, no personal love or hate, clear reasoning.

Reply:
HasOne is fine - you will write has_one :)

Reply:
What would be the use case for converting PascalCase to snake_case?

Reply:
There are at least 3 reasons:
- clear rule how to name it (otherwise you are inconsistent, this is actually what Microsoft and other big companies very strictly does)
- you can use it for autonaming (not only with snake_case, but also for title - with this approach, you can simply identify words easily, then you can replace the placed betteen words with anything you want - "_", "-", spaces, ...)
- Faster to read (because of previous point - you can also tell much faster where does what word start)
- and probably many others :)

So explanation phase it done. Can you now update your refactoring PRs with this naming so we can continue?